src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,44 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int VALID_LOGIN_LENGTH = 6;
+    public static final int VALID_PASSWORD_LENGTH = 6;
+    public static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        Objects.requireNonNull(user);
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegisterException("User with " + user.getLogin()
+                    + " as a login already exist");
+        }
+
+        if (user.getLogin().length() < VALID_LOGIN_LENGTH) {
+            throw new RegisterException(
+                    "The login must have at least " + VALID_LOGIN_LENGTH
+                            + " symbols, but you have: "
+                            + user.getLogin().length());
+        }
+
+        if (user.getPassword().length() < VALID_PASSWORD_LENGTH) {
+            throw new RegisterException(
+                    "The password must have at least " + VALID_PASSWORD_LENGTH
+                            + " symbols, but you have: "
+                            + user.getPassword().length());
+        }
+
+        if (user.getAge().compareTo(VALID_AGE) < 0) {
+            throw new RegisterException(
+                    "The age must have at least " + VALID_AGE
+                            + " years old, but yours is : "
+                            + user.getAge());
+        }
```

Comment:
I think you can split this ifs in different methods.

Reply:
I don't see the point to separate them, because its uses only once in code

Reply:
Code duplication isn't the only reason for creating separate methods.
You method register should just register user and other methods could check or validate some parameters. Also, there are many if statements in one method.

In addition, it's better to check if login, password and age are null. 
In such case you will have more if statements and this will also cause separate methods to be used.